### DISCUSSION TITLE
Resize adjacent windows feels broken

### DISCUSSION THREAD

**Discussion by austincondiff** [rxhanson/RectanglePro-Community#410] [Ideas]
Resize adjacent windows is very janky and doesn't work as expected. Consider the following:

- Resize adjacent windows doesn't work if the user has a "gap between windows" set
- Resizing of the other window only happens on mouse up which feels like it's broken and unnatural 
- When a window is in focus while in a layout, dragging to resizing works, however if the other window that was resized is then focused and resized, the original window doesn't resize in response.
- When the cursor is over a resizable area, show a transparent rectangle along the edge, letting the user know that this edge is recognized by Rectangle Pro and will resize other windows when it is resized. This ghost rectangle might be 10 pixels in width and If there is a gap set, this rectangle would fill that gap instead.

Once these are resolved, this feature will feel much better to use.

In full disclosure I was about to purchase Rectangle Pro for this feature alone. However after trying it, these things prevent me from wanting to.

**Comment by rxhanson** [comment]
Thanks for the feedback. 

> Resize adjacent windows doesn't work if the user has a "gap between windows" set

How large are the gaps that you're using?

> Resizing of the other window only happens on mouse up which feels like it's broken and unnatural

There are tradeoffs here, and with what I have available to me as a 3rd party developer, this is still my preferred implementation. Resizing windows from the left edge continuously makes the window edge bounce around. In my testing, it also makes it hard to wrangle multiple windows consistently, for example if you have a 2x2 grid. 

> When a window is in focus while in a layout, dragging to resizing works, however if the other window that was resized is then focused and resized, the original window doesn't resize in response.

I've had other reports of this and will be working on it for an upcoming release.

**Reply by austincondiff** [reply]
I've added one more point to my original list you may also consider.

**Reply by austincondiff** [reply]
> How large are the gaps that you're using?

20 pixels

> Resizing windows from the left edge continuously makes the window edge bounce around.

I see, which would feel equally as janky. This must be because the window is getting resized after on an interval. There must be a way to update the size in a more immediate manner. Even if it lags behind a frame, as long as it is smooth, thats all that matters. I understand the problem, however this would be ideal if you can find a solution.

> I've had other reports of this and will be working on it for an upcoming release.

Perfect!

**Reply by rxhanson** [reply]
I went ahead and pushed out v3.0.15 with some adjustments here, please try it out and let me know how it goes.
https://rectangleapp.com/pro/versions
